### DISCUSSION TITLE
npm start not creating http routes

### DISCUSSION THREAD

**Discussion by waxidiotic** [beginner-corp/begin-community#74] [Support] [answered]
I edited my `app.arc` file to add two http routes - `get /cookies` and `post /cookies`. After which, I ran `npm start` and was under the impression that the `src/http/get-cookies` and `src/http/post-cookies` folders and starter code would be created automatically.

Arc file:
```
@app
slack-cookie-bot

@http
get /cookies
post /cookies

@tables
data
  scopeID *String
  dataID **String
  ttl TTL
```

Am I doing something wrong? Should I create the folders manually?

**Comment by brianleroux** [comment, marked as answer]
hey @waxidiotic sorry for the confusion! if you install the architect cli (`npm i -g @architect/architect`) you can generate these files with `arc init`

**Reply by ryanblock** [reply]
Without installing Arc, you can also create a preferences.arc file and enable autocreate: https://arc.codes/docs/en/reference/prefs.arc/create

**Reply by waxidiotic** [reply]
Awesome. That worked. Thanks!

FYI though - the docs do say they are automatically generated for you out-of-the-box.

https://docs.begin.com/en/http-functions/provisioning
